From: Ilya Dryomov <idryomov@gmail.com>
Date: Mon, 16 Oct 2017 10:32:50 +0200
Subject: ceph: -EINVAL on decoding failure in ceph_mdsc_handle_fsmap()
Git-commit: 76bd6ec4988065d39983ba8e93bb738313f68050
Patch-mainline: v4.15-rc1
References: FATE#324714

Don't set ->mdsmap_err to -ENOENT unconditionally, and drop unneeded
return statement while at it.

Signed-off-by: Ilya Dryomov <idryomov@gmail.com>
Acked-by: Luis Henriques <lhenriques@suse.com>
---
 fs/ceph/mds_client.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/fs/ceph/mds_client.c
+++ b/fs/ceph/mds_client.c
@@ -3878,14 +3878,14 @@ void ceph_mdsc_handle_fsmap(struct ceph_
 		goto err_out;
 	}
 	return;
+
 bad:
 	pr_err("error decoding fsmap\n");
 err_out:
 	mutex_lock(&mdsc->mutex);
-	mdsc->mdsmap_err = -ENOENT;
+	mdsc->mdsmap_err = err;
 	__wake_requests(mdsc, &mdsc->waiting_for_map);
 	mutex_unlock(&mdsc->mutex);
-	return;
 }
 
 /*
